src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -129,8 +127,22 @@ class TaskReplaceActor(
       instancesStarted -= 1
       launchInstances().pipeTo(self)
 
+    // An old instance terminated out of band and was not yet chosen to be decommissioned or stopped
+    // the instance will be rescheduled with the new version so we don't have to do anything
+    case InstanceChanged(id, _, `pathId`, condition, instance) if oldInstanceIds(id) && instance.state.goal == Goal.Running =>
+      logger.info(s"Old instance $id became $condition during an upgrade but still has goal Running. A new task will be launched but it will retain the instanceId.")
```

Comment:
That is not true. below we set the goal to `Stopped` or `Decomissioned` so it won't retain the instance id. Also, at least for ephemeral instances we will start a new instance so the id has to be different.

In any case, why are we checking this out of band termination?

Reply:
@jeschkies has a great point; if we set the ephemeral instance goal to decommissioned then a new instance id will be allocated.

Reply:
I have adjusted the log line, good catch 👍